source/tutorials/hpc.rst

```diff
@@ -0,0 +1,693 @@
+
+      Example output:
+
+      .. code-block:: console
+
+         hpc-worker4: NodeName=hpc-worker4 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=1915
+         hpc-worker4: UpTime=0-01:23:28
+         hpc-worker3: NodeName=hpc-worker3 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=1663
+         hpc-worker3: UpTime=0-01:33:41
+         hpc-worker2: NodeName=hpc-worker2 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=721
+         hpc-worker2: UpTime=0-01:34:56
+         hpc-worker1: NodeName=hpc-worker1 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=721
+         hpc-worker1: UpTime=0-01:39:21
+
+   #. Get their IP addresses.
+
+      .. code-block:: bash
+
+         pdsh -w ^worker-nodes-list "networkctl status | awk 'FNR == 2 {print $2}'"
+
+      Example output:
+
+      .. code-block:: console
+
+         hpc-worker4:          Address: 192.168.1.23 on eth0
+         hpc-worker3:          Address: 192.168.1.22 on eth0
+         hpc-worker2:          Address: 192.168.1.21 on eth0
+         hpc-worker1:          Address: 192.168.1.20 on eth0
+
+   #. :command:`sudoedit` the :file:`slurm.conf` file.  Append each worker node 
+      information plus its IP address, but without the `UpTime`, under the 
+      `COMPUTE NODES` section.
+
+      .. tip::
+
+         It is strongly recommended to set the `RealMemory` value for each 
+	 worker node slightly below, say 90%, what was reported by 
+	 :command:`slurmd -C` 
+	 in case some memory gets use by some processes, which would 
+	 cause slurm not to make a node not available due to its memory 
+	 resource falling below the stated value in the configuration file.    
+
+      Here's an example with four worker nodes:
+
+      .. code-block:: console
+
+         # 
+         # COMPUTE NODES (mode detailed values reported by "slurmd -C" on each node)
+         NodeName=hpc-worker1 NodeAddr=192.168.1.20 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=648
+         NodeName=hpc-worker2 NodeAddr=192.168.1.21 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=648
+         NodeName=hpc-worker3 NodeAddr=192.168.1.22 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=1497
+         NodeName=hpc-worker4 NodeAddr=192.168.1.23 CPUs=1 Boards=1 SocketsPerBoard=1 CoresPerSocket=1 ThreadsPerCore=1 RealMemory=1723
+
+   #. Create partitions.
+
+      A slurm partition is basically the grouping of worker nodes.  
+      Give each partition a name and decide which worker node(s) go belong to 
+      it.
+
+      For example:
+   
+      .. code-block:: console
+      
+         PartitionName=workers Nodes=hpc-worker1,hpc-worker2,hpc-worker3, hpc-worker4 Default=YES MaxTime=INFINITE State=UP
+         PartitionName=debug Nodes=hpc-worker1,hpc-worker3 MaxTime=INFINITE State=UP
+
```

Comment:
Formatting issue: This is a bit misleading, should be split in two separate examples. User can use either of the two partitions definitions, but not both of them. Also an extra space following ',' in front of hpc-worker4

Reply:
writer error!  fixed.